lua/haskell-snippets/table-util.lua

```diff
@@ -0,0 +1,35 @@
+---@mod haskell-snippets.table-util
+
+---@brief [[
+--- Extra utilities related to Lua tables.
+---@brief ]]
+
+local M = {}
+
+---Concatenates tables.
+---
+---@return table
+function M.concat(...)
```

Comment:
This can be replaced with [`vim.tbl_extend('force', ...)`](https://neovim.io/doc/user/lua.html#vim.tbl_extend()).

Reply:
I don't see it. Could you elaborate?

`vim.tbl_extend` works like a dictionary update and not a concatenater. For example, `vim.tbl_extend('force', {'a'}, {'b', 'c'})` returns `{'b', 'c'}`, but I with `concat` I want `{'a', 'b', 'c'}`.

I've searched for a concatenating function inside Neovim's Lua API and couldn't find any.

Reply:
Oh sorry, my bad.
In that case, you can use [`vim.list_extend`](https://neovim.io/doc/user/lua.html#vim.list_extend()). It mutates the first argument, but that doesn't matter in this case.

Reply:
Thanks.